src/voigt.jl

```diff
@@ -105,25 +124,60 @@ julia> tovoigt!(x, T; offset=1)
 ```
 """
 function tovoigt! end
-Base.@propagate_inbounds function tovoigt!(v::AbstractVector, A::Tensor{2, dim}; offset::Int=0, order=DEFAULT_VOIGT_ORDER[dim]) where {dim}
+Base.@propagate_inbounds function tovoigt!(v::AbstractVector, A::Tensor{2}; offset::Int=0, order=nothing)
+    _tovoigt!(v, A, order; offset=offset)
+end
+Base.@propagate_inbounds function tovoigt!(v::AbstractMatrix, A::Tensor{4}; offset_i::Int=0, offset_j::Int=0, order=nothing)
+    _tovoigt!(v, A, order; offset_i=offset_i, offset_j=offset_j)
+end
+Base.@propagate_inbounds function tovoigt!(v::AbstractVector{T}, A::SymmetricTensor{2}; offdiagscale=one(T), offset::Int=0, order=nothing) where T
+    _tovoigt!(v, A, order; offdiagscale=offdiagscale, offset=offset)
+end
+Base.@propagate_inbounds function tovoigt!(v::AbstractMatrix{T}, A::SymmetricTensor{4}; offdiagscale=one(T), offset_i::Int=0, offset_j::Int=0, order=nothing) where T
+    _tovoigt!(v, A, order; offdiagscale=offdiagscale, offset_i=offset_i, offset_j=offset_j)
+end
```

Comment:
I intentionally kept all of these different versions because they take different kwargs. 

The old behavior was that e.g. a `offdiagscale` is not applicable to `Tensor` and thus you would get an informative method error from `tovoigt!` (=the function the user called) when using it. This suggestion still throws a method error for `_tovoigt!` but that is an internal function - a bit more difficult to pick up on what you did wrong as a user in that case.

E.g. this is a helpful error I think:

Reply:
That's a good point! 
In that case, should we remove the defaults for offdiagscale and offsets in the internal function to avoid specifying these twice?

Reply:
That didn't work out so nicely, so let's keep it as it is wrt that!